aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/SdkConfigDecorator.kt

```diff
@@ -12,11 +12,56 @@ import software.amazon.smithy.rust.codegen.client.smithy.generators.config.Confi
 import software.amazon.smithy.rust.codegen.client.smithy.generators.config.ServiceConfig
 import software.amazon.smithy.rust.codegen.core.rustlang.RustModule
 import software.amazon.smithy.rust.codegen.core.rustlang.Writable
+
+import software.amazon.smithy.rust.codegen.core.rustlang.join
+
 import software.amazon.smithy.rust.codegen.core.rustlang.rust
 import software.amazon.smithy.rust.codegen.core.rustlang.rustTemplate
 import software.amazon.smithy.rust.codegen.core.rustlang.writable
 import software.amazon.smithy.rust.codegen.core.smithy.RuntimeConfig
 import software.amazon.smithy.rust.codegen.core.smithy.RustCrate
+import software.amazon.smithy.rust.codegen.core.smithy.customize.DetachedSection
+import software.amazon.smithy.rust.codegen.core.smithy.customize.Section
+
+/**
+ * Section enabling linkage between `SdkConfig` and <service>::Config
+ */
+object SdkConfigSection : DetachedSection<SdkConfigSection.CopySdkConfigToClientConfig>("SdkConfig") {
+    /**
+     * [sdkConfig]: A reference to the SDK config struct
+     * [serviceConfigBuilder]: A reference (owned) to the `<service>::config::Builder` struct.
+     *
+     * Each invocation of this section MUST be a complete statement (ending with a semicolon), e.g:
+     * ```
+     * rust("${section.serviceConfigBuilder}.set_foo(${section.sdkConfig}.foo());")
+     * ```
+     */
+    data class CopySdkConfigToClientConfig(val sdkConfig: String, val serviceConfigBuilder: String) :
+        Section("CopyConfig")
+}
+
+/**
+ * SdkConfig -> <service>::Config for settings that come from generic smithy
+ */
+class GenericSmithySdkConfigSettings : ClientCodegenDecorator {
+    override val name: String = "GenericSmithySdkConfigSettings"
+    override val order: Byte = 0
+
+    override fun extraSections(codegenContext: ClientCodegenContext): List<Pair<DetachedSection<*>, (Section) -> Writable>> =
+        listOf(
+            SdkConfigSection.create { section ->
+                writable {
+                    rust(
+                        """
+                        ${section.serviceConfigBuilder}.set_retry_config(${section.sdkConfig}.retry_config().cloned());
+                        ${section.serviceConfigBuilder}.set_timeout_config(${section.sdkConfig}.timeout_config().cloned());
+                        ${section.serviceConfigBuilder}.set_sleep_impl(${section.sdkConfig}.sleep_impl());
```

Comment:
@jdisanti refers to this stuff as "Resiliency"
https://github.com/awslabs/smithy-rs/blob/main/codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/customizations/ResiliencyConfigCustomization.kt

It's worth considering if we should move this code into a similarly-named package in AWS codegen.